presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/EnvironmentListener.java

```diff
@@ -161,4 +163,24 @@ public void onContainerStopped(DockerContainer container, InspectContainerRespon
             }
         };
     }
+
+    static EnvironmentListener logCopyingListener(Optional<Path> logBaseDir)
```

Comment:
make `logBaseDir` non-empty and instead do not create listener in code flow when now you create it with `Optional.empty()`

Reply:
I think it will make code less readable

Reply:
But more proper imo. You can create some external helper tooling to construct proper listener set. Logging listener should not allow empty log pth

Reply:
Fixed.